.circleci/config.yml

```diff
@@ -0,0 +1,187 @@
+# (c) Copyright 2019 Hewlett Packard Enterprise Development LP
+#
+# Check https://circleci.com/docs/2.0/language-python/ for more details
+#
+version: 2.1
+
+workflows:
+  build and publish:
+    jobs:
+      - build-container
+      - verify-container:
+          requires:
+            - build-container
+      - publish-container-hub:
+          requires:
+            - verify-container
+      - publish-container-quay:
+          requires:
+            - verify-container
+
+defaults:
+  - restore_cache: &python_deps_restore_cache
+      keys:
+        - &python_deps_cache_key v1-dependencies-{{ checksum "requirements.txt" }}-{{ checksum "test-requirements.txt" }}
+        - v1-dependencies-
+
+  - save_cache: &python_deps_save_cache
+      paths:
+        - "~/.tox"
+        - "~/.cache/pip"
+      key: *python_deps_cache_key
+
+jobs:
+
+  build-container:
+    docker:
+      - image: circleci/python:3.6
+
+    working_directory: ~/repo
+
+    steps:
+      - checkout
+      - setup_remote_docker
+      # Download and cache dependencies
+      - restore_cache: *python_deps_restore_cache
+
+      - run:
+          name: Setup Environment Variables
+          command: |
+            echo 'export PATH="~/.local/bin:${PATH}"' >> $BASH_ENV
+      - run:
+          name: Install Project Locally
+          command: |
+            pip install --user -e .
+      - run:
+          name: Build Image From Local Install
+          command: |
+            windlass --debug --build-only
+
+      - run:
+          name: Archive Docker Image
+          command: |
+            docker save -o image.tar zing/windlass
+
+      - save_cache: *python_deps_save_cache
+      - persist_to_workspace:
+          root: ~/repo
+          paths:
+            - image.tar
+
+  verify-container:
+    docker:
+      - image: circleci/python:3.6
+
+    working_directory: ~/repo
+
+    steps:
+      - attach_workspace:
+          at: /tmp/workspace
```

Comment:
q: Why not attaching at ~/repo here?

Reply:
Was just looking to keep any files added to the workspace separate from a clean source checkout in case in the future something was added that might alter the behaviour